File: masterclass/Demomvp/app/src/main/java/com/example/demo_mvp/home/HomeUiData.kt
package com.example.demo_mvp.home

import androidx.annotation.ColorInt
import androidx.annotation.DrawableRes

data class Transaction(
    val id: String,
    val title: String,
    val category: String,
    val amount: String,
    val time: String,
    @param:DrawableRes val iconRes: Int,
    @param:ColorInt val badgeColor: Int,
    @param:DrawableRes val rowBackgroundRes: Int,
)

data class HomeUiData(
    val userName: String,
    val activityValues: List<Int>,
    val activityLabels: List<String>,
    val salesLastWeek: String,
    val revenueLastWeek: String,
    val transactions: List<Transaction>,
)

File: masterclass/Demo-mvvm-iOS/Demo-mvvm-iOS/Home/HomeUiData.swift
import SwiftUI

struct Transaction: Identifiable, Equatable {
    let id: String
    let title: String
    let category: String
    let amount: String
    let time: String
    let iconSystemName: String
    let badgeColor: Color
    let rowBackgroundColor: Color
}

struct HomeUiData: Equatable {
    let userName: String
    let activityValues: [Int]
    let activityLabels: [String]
    let salesLastWeek: String
    let revenueLastWeek: String
    let transactions: [Transaction]
}
